### PR TITLE
Enable SP7 testing

### PR DESCRIPTION
**Pull request by dcermak**
This will not work properly before we have the SLE 15 SP7 BCI repo

This fixes https://github.com/SUSE/BCI-tests/issues/545

### CODE REVIEW

**Comment by dirkmueller** [comment]
@dcermak rebase? we need to get this in soonish when 15.7 development starts (I'd be fine with testing all of base/minimal for now, similar to how 16.0 is enabled)

File: tests/test_all.py [hunk 1]
```diff
@@ -376,6 +376,13 @@ def test_no_orphaned_packages(container_per_test: ContainerData) -> None:
 @pytest.mark.parametrize(
     "container", CONTAINERS_WITH_ZYPPER_AS_ROOT, indirect=True
 )
+@pytest.mark.xfail(
```
Comment (alexandrevicenzi) [inline tests/test_all.py:379]: This seems to break tests on 15.4.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -167,6 +165,14 @@ jobs:
             os_version: "16.0"
           - toxenv: kernel_module
             os_version: "16.0"
+          - toxenv: all
+            os_version: "15.7"
```
Comment (dirkmueller) [inline .github/workflows/ci.yaml:169]: although don't have a strong opinion, I think sorting these before the "16.0" section would be slightly more logical.

File: tests/test_all.py (hunk 1 above)
Comment (dirkmueller) [reply tests/test_all.py:379]: yeah, because they're run on a non-suse host so the container-suseconnect integration is not working. rather than adding a quirk in the tests for this, maybe we should simply not define a repo like in the 16.0 case?

File: bci_tester/data.py [hunk 3]
```diff
@@ -519,7 +519,7 @@ def create_BCI(
 )
 PYTHON311_CONTAINER = create_BCI(
     build_tag="bci/python:3.11",
-    available_versions=_DEFAULT_NONBASE_OS_VERSIONS,
+    available_versions=["15.6", "tumbleweed"],
```
Comment (dirkmueller) [inline bci_tester/data.py:522]: python 311 is still available with SP7, why this change?

**Review by dirkmueller — APPROVED** [review]
